### PR TITLE
Extend Github plugin to support non-PR builds

### PR DESCRIPTION
**Pull request by sandro-h**

### CODE REVIEW

**Comment by sandro-h** [comment]
@dodie, @hosszubalazs: the renaming of folders causes quite some diff noise. I suggest to select the "Extend Github plugin to support non-PR builds" commit in the diff view to see the actual code changes.

**Comment by hosszubalazs** [comment]
Why is this PR analyyed as a branch instead of a PR in Sonar?
https://sonarcloud.io/dashboard?branch=github_status&id=adnovum_gong-notifier

**Comment by sandro-h** [comment]
> Why is this PR analyyed as a branch instead of a PR in Sonar?

No idea why it didn't work the first time, but now that I reran the build it treated it correctly.

File: gong-notifier-github-status/README.md [hunk 1]
```diff
@@ -61,29 +60,32 @@ There are two ways to configure the access token:
 1. As a secret variable `GONG_STATUS_AUTH_TOKEN`. The variable must be defined on the pipeline, not on a stage or job.
```
Comment (dodie) [inline gong-notifier-github-status/README.md:60]: The 2nd way of configuring the plugin suggests that if the material is configured with credentials, those credentials will be reused for the feedback. For private repositories, this seems the way to go. Do I understand correctly that the explicit `GONG_STATUS_AUTH_TOKEN` setting is for GitHub repositories that can be accessed anonymously?

Also, based on the above, is it true that just by installing the plugin one can get GitHub build feedback for private repositories with no additional configuration?

File: gong-notifier-github-pr-status/src/main/java/ch/adnovum/gong/notifier/github/pr/status/GithubStatusHelper.java [hunk 2]
```diff
@@ -121,16 +167,35 @@ public static String getRepoFromUrl(String gitUrl) {
...
+				.filter(m -> GIT_MATERIAL_TYPE.equals(m.type))
+				.filter(m -> ghInfo.url.equals(m.attributes.get("url")))
+				.filter(m -> m.attributes.containsKey(ENCRYPTED_PASSWORD_KEY) || m.attributes.containsKey(PASSWORD_KEY))
+				.findFirst()
+				.map(GithubStatusHelper::gitMatToEnvVar);
+	}
+
+	private static EnvironmentVariable gitMatToEnvVar(PipelineConfig.Material mat) {
+		EnvironmentVariable var = new EnvironmentVariable();
+		var.name = "password";
```
Comment (dodie) [inline gong-notifier-github-pr-status/src/main/java/ch/adnovum/gong/notifier/github/pr/status/GithubStatusHelper.java]: Would it be possible to also reuse `PASSWORD_KEY` here?

File: gong-notifier-github-status/README.md (hunk 1 above)
Comment (sandro-h) [reply gong-notifier-github-status/README.md:60]: > Do I understand correctly that the explicit GONG_STATUS_AUTH_TOKEN setting is for GitHub repositories that can be accessed anonymously?

Correct, or for any other reason you might want a different identity to do the status updates.

> Also, based on the above, is it true that just by installing the plugin one can get GitHub build feedback for private repositories with no additional configuration?

Yep that's the idea.

**Pull request merged**